source/parser.hera

```diff
@@ -3536,6 +3538,9 @@ JSXAttributeValue
 JSXChildren
   ( __ JSXChild )*
 
+JSXNestedChildren
+  PushIndent ( Nested JSXChild )+ PopIndent
```

Comment:
One quirk of tracking indentation state is that `PopIndent` must always be called if `PushIndent` succeeds. This means that you need to use `*` and `return $skip` in the handler if the length is zero to make sure the indentation state stack resets.

Reply:
Thanks; I wondered why that code was there.

On another point, I found that replacing `( Nested JSXChild )` with `JSXNestedChild` and a separate rule:
```hera
NestedJSXChild
  Nested JSXChild
```
broke the parsing.  Any ideas why this would occur?

Reply:
I think it might have to do with caching in the parser. In `source/main` the exit handler there is a big list of rules that shouldn't be cached. Since `JSXChild` includes `__` if it is cached it might override indentation after backtracking. Let me know if that is making sense.